config/setup/git.go

```diff
@@ -20,13 +21,31 @@ func Git(c *Controller) (middleware.Middleware, error) {
 		return nil, err
 	}
 
-	c.Startup = append(c.Startup, func() error {
-		// Start service routine in background
-		git.Start(repo)
+	// If a HookUrl is set, we switch to event based pulling.
+	// Install the url handler
+	if repo.HookUrl != "" {
 
-		// Do a pull right away to return error
-		return repo.Pull()
-	})
+		err := repo.Pull()
```

Comment:
Now we're talkin'! However, this should be in a Startup function, identical to the one on line 40, except without starting the git service that does the interval pulling, obviously. We'd rather not pull the repo during parse-time. :)